internal/configs/config.go

```diff
@@ -460,6 +460,27 @@ func (c *Config) addProviderRequirements(reqs getproviders.Requirements, qualifs
 		})
 		reqs[fqn] = nil
 	}
+	for _, rc := range c.Module.EphemeralResources {
```

Comment:
We might want to consider breaking this block into it's own function, am I reading it right that it's the same for resource/data/ephemeral?

Reply:
Thanks a lot! Reworked in #2876 (bbf8c18c0e56baf7041c71dc2b96590bec399b79)